examples/C++/using_installed_STIR/demo_create_image.cxx

```diff
@@ -0,0 +1,60 @@
+#include "stir/Shape/EllipsoidalCylinder.h"
+#include "stir/PatientPosition.h"
+#include "stir/ImagingModality.h"
+#include "stir/CartesianCoordinate3D.h"
+#include "stir/IndexRange3D.h"
+#include "stir/Succeeded.h"
+#include "stir/IO/write_to_file.h"
+#include "stir/VoxelsOnCartesianGrid.h"
+#include <iostream>
+
+int main()
+{
+
+  std::string output_filename("test.hv");
+  int output_image_size_x = 100;
+  int output_image_size_y = 104;
+  int output_image_size_z = 20;
+  float output_voxel_size_x = 5.F;
+  float output_voxel_size_y = 4.F;
+  float output_voxel_size_z = 3.F;
+  double image_duration = 100.; // secs
+  double rel_start_time = 1; //secs
+
+  auto exam_info_sptr = std::make_shared<stir::ExamInfo>(stir::ImagingModality::PT);
+  {
+    std::vector<double> start_times(1, rel_start_time);
+    std::vector<double> durations(1, image_duration);
+    stir::TimeFrameDefinitions frame_defs;
+    frame_defs.set_num_time_frames(1);
+    frame_defs.set_time_frame(1, rel_start_time, image_duration);
+    exam_info_sptr->set_time_frame_definitions(frame_defs);
+  }
+  stir::VoxelsOnCartesianGrid<float>
+          image(exam_info_sptr,
+                        stir::IndexRange3D(0,output_image_size_z-1,
+                                     -(output_image_size_y/2),
+                                     -(output_image_size_y/2)+output_image_size_y-1,
+                                     -(output_image_size_x/2),
+                                     -(output_image_size_x/2)+output_image_size_x-1),
+                        stir::CartesianCoordinate3D<float>(0,0,0),
+                        stir::CartesianCoordinate3D<float>(output_voxel_size_z,
+                                                     output_voxel_size_y,
+                                                     output_voxel_size_x));
+
+  const auto centre =
+    image.get_physical_coordinates_for_indices(stir::make_coordinate(output_image_size_z/2,0,0));
+  stir::EllipsoidalCylinder shape(40.F, 30.F, 20.F, centre);
+  shape.construct_volume(image, stir::make_coordinate(2,2,2));
+
+  try
+    {
+      stir::write_to_file(output_filename, image);
+      stir::info("Image written as " + output_filename);
+    }
+  catch (...)
+    {
+      return EXIT_FAILURE;
+    }
+  return EXIT_SUCCESS;  
```

Comment:
```suggestion
  try
    {
      stir::write_to_file(output_filename, image);
    }
  catch (...)
    {
      stir::info("Failed to write image to file " + output_filename);
      return EXIT_FAILURE;
    }
  stir::info("Image written as " + output_filename);
  return EXIT_SUCCESS;  
```

Reply:
same result really. if `write_to_file` throws, the `info` will never be  executed.

Reply:
True, minor point, ignore it. I am learning error handling with golang and I generally only `try` one thing at a time, especially with `catch (...)`. In this case it does not matter as it is the end of the program.